lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -192,5 +185,51 @@ def pretty_status
     def kubectl
       @kubectl ||= Kubectl.new(namespace: @namespace, context: @context, logger: @logger, log_failure_by_default: false)
     end
+
+    class Event
+      JSONPATH_FIELDS =
+        "{.involvedObject.kind}\t{.involvedObject.name}\t{.count}\t{.lastTimestamp}\t{.reason}\t{.message}"
+      JSONPATH_SEPARATOR = "ENDEVENT\nBEGINEVENT"
+
+      def self.jsonpath_for(name)
+        %({range .items[?(@.involvedObject.name=="#{name}")]}#{JSONPATH_FIELDS}#{JSONPATH_SEPARATOR}{end})
```

Comment:
Anyone with jsonpath experience know a way to compose this less sketchily? (or any other reasonably efficient way to get relevant events)

Relevant:
* https://kubernetes.io/docs/user-guide/jsonpath/
* [issue: jsonpath template output should be json](https://github.com/kubernetes/kubernetes/issues/16707)
* [issue: kubectl: no way to list events about a specific object](https://github.com/kubernetes/kubernetes/issues/11994)
* [issue: Unable to use kubectl get with a fieldSelector](https://github.com/kubernetes/kubernetes/issues/14129)